documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,947 @@
+# Migrate from v1 to v2 of the AWS IoT SDK for Java
+
+The AWS IoT Device SDK for Java v2 is a major rewrite of the v1 SDK code base built on top of Java 8+. It includes many updates,
+such as improved consistency, ease of use, more detailed information about client status, and an offline operation queue
+control. This guide describes the major features that are new in the v2 SDK, and provides guidance on how to migrate
+your code to v2 from v1 of the AWS IoT SDK for Java.
+
+> [!NOTE]  
+> If you can't find the information you need in this guide, visit the [How to get help](#how-to-get-help) section for more help and guidance.
+
+* [What's new in AWS IoT Device SDK for Java v2](#whats-new-in-aws-iot-device-sdk-for-java-v2)
+* [How to get started with AWS IoT Device SDK for Java v2](#how-to-get-started-with-aws-iot-device-sdk-for-java-v2)
+    * [Package name change](#package-name-change)
+    * [MQTT protocol](#mqtt-protocol)
+    * [Client builder](#client-builder)
+    * [Connection types and features](#connection-types-and-features)
+    * [Lifecycle events](#lifecycle-events)
+    * [Publish](#publish)
+    * [Subscribe](#subscribe)
+    * [Unsubscribe](#unsubscribe)
+    * [Client stop](#client-stop)
+    * [Client shutdown](#client-shutdown)
+    * [Reconnects](#reconnects)
+    * [Offline operations queue](#offline-operations-queue)
+    * [Operation timeouts](#operation-timeouts)
+    * [Logging](#logging)
+    * [Client for AWS IoT Device Shadow](#client-for-aws-iot-device-shadow)
+    * [Client for AWS IoT Jobs](#client-for-aws-iot-jobs)
+    * [Client for AWS IoT fleet provisioning](#client-for-aws-iot-fleet-provisioning)
+    * [Example](#example)
+* [How to get help](#how-to-get-help)
+* [Appendix](#appendix)
+    * [MQTT5 features](#mqtt5-features)
+
+## What's new in AWS IoT Device SDK for Java v2
+
+* The v2 SDK client is truly async. Operations return `CompletableFuture` objects. Blocking calls can be emulated by waiting
+for the returned `CompletableFuture` object to be resolved.
+* The v2 SDK provides implementation for MQTT5 protocol, the next step in evolution of the MQTT protocol.
```

Comment:
Trivial: "The v2 SDK provides an MQTT5 Client which implements the MQTT5 protocol, the next significant evolution of the MQTT protocol."